paddle/phi/core/distributed/auto_parallel/dist_attr.cc

```diff
@@ -348,5 +348,69 @@ bool TensorDistAttr::empty() const {
   return process_mesh_.empty() || dims_mapping_.empty();
 }
 
+std::vector<std::shared_ptr<PlacementStatus>> TensorDistAttr::to_placement()
+    const {
+  auto ndim = process_mesh_.ndim();
+  std::vector<std::shared_ptr<PlacementStatus>> placement(
+      ndim, std::make_shared<ReplicatedStatus>());
+  for (size_t i = 0; i < dims_mapping_.size(); ++i) {
+    if (dims_mapping_[i] != -1) {
+      PADDLE_ENFORCE_LT(
+          dims_mapping_[i],
+          ndim,
+          errors::InvalidArgument(
+              "Split axis %ld can not exceed the ndim of process_mesh %ld",
+              dims_mapping_[i],
+              ndim));
+      placement[dims_mapping_[i]] = std::make_shared<ShardStatus>(i);
```

Comment:
why shared_ptr ？ the return placement is supposed for temporary usage and should not be hold by others.